clients/da-vinci-client/src/test/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackendTest.java

```diff
@@ -191,18 +193,36 @@ public void testIsolatedIngestionNotifierAsyncCompletionHandling() {
     VeniceNotifier ingestionNotifier = mock(VeniceNotifier.class);
     VeniceConfigLoader configLoader = mock(VeniceConfigLoader.class);
     VeniceStoreVersionConfig storeVersionConfig = mock(VeniceStoreVersionConfig.class);
+    MainIngestionMonitorService mainIngestionMonitorService = mock(MainIngestionMonitorService.class);
     ExecutorService executor = Executors.newFixedThreadPool(10);
+
+    when(backend.getMainIngestionMonitorService()).thenReturn(mainIngestionMonitorService);
     when(backend.getCompletionHandlingExecutor()).thenReturn(executor);
     when(backend.getIsolatedIngestionNotifier(any())).thenCallRealMethod();
     when(backend.getConfigLoader()).thenReturn(configLoader);
+
     String topic = "topic_v1";
     when(configLoader.getStoreConfig(topic)).thenReturn(storeVersionConfig);
-    when(backend.isTopicPartitionIngesting(topic, 0)).thenReturn(false);
-    when(backend.isTopicPartitionIngesting(topic, 1)).thenReturn(true);
+    when(backend.isTopicPartitionHosted(topic, 0)).thenReturn(false);
+    when(backend.isTopicPartitionHosted(topic, 1)).thenReturn(true);
+    when(backend.isTopicPartitionHosted(topic, 2)).thenReturn(true);
     backend.getIsolatedIngestionNotifier(ingestionNotifier).completed(topic, 0, 123L, "", Optional.empty());
-    verify(backend, times(0)).getCompletionHandlingExecutor();
+    TestUtils.waitForNonDeterministicAssertion(5, TimeUnit.SECONDS, true, () -> {
+      verify(backend, times(0)).getCompletionHandlingExecutor();
+    });
```

Comment:
`times(0)` doesn't need `waitForNonDeterministicAssertion`

Reply:
Yeah that's true, I was doing cmd+c/v :)